parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetRecordReader.java

```diff
@@ -188,7 +189,11 @@ private void checkDeltaByteArrayProblem(FileMetaData meta, Configuration conf, B
       // this is okay if not using DELTA_BYTE_ARRAY with the bug
       Set<Encoding> encodings = new HashSet<Encoding>();
       for (ColumnChunkMetaData column : block.getColumns()) {
-        encodings.addAll(column.getEncodings());
+        try {
+          encodings.addAll(column.getEncodings());
+        } catch (HiddenColumnException e) {
+          LOG.warn("Bypass column {} because it is hidden column so encodings cannot be read", column);
+        }
```

Comment:
Sounds good. Please remove the InternalFileDecryptor.java changes, and I'll merge this pr.